src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -442,19 +511,19 @@ private void PreProcessUserFunctionStatement(ParseNode node)
 
                 if (userFuncObject.IsSystemLock())
                 {
-                    AddOpcode(new OpcodePush(userFuncObject.PointerIdentifier));
+                    AddOpcode(new OpcodePush(userFuncObject.ScopelessPointerIdentifier));
```

Comment:
to keep the names unique when looking up the function, the same identifier name gets different values based on the scope ID it was in, so function Foo in global scope gets stored as having identifier "foo`0" and a foo function in, say, the 23rd brace section encountered in the compiler gets stored as identifier "foo`23".  The ScopelessPointerIdentifier strips off that trailing number again to just give you "foo".